src/gurt/dtm.c

```diff
@@ -0,0 +1,425 @@
+/**
```

Comment:
Not sure why this didn't show as a diff.  This file is moved from src/client/dfuse/dfuse_da.c.  The only changes are function, struct, and member names.